src/Hamiltonians/Molecular.jl

```diff
@@ -0,0 +1,201 @@
+    addr::A # Contains address Psi_i, provided by operator_column
+    op::O   # Represent Hamiltonian itself, provided by operator_column
+    diag::T # < Psi_i | Psi_i >, calculated by diagonal_element
+    ods::OD # Vector [ < Psi_j | Psi_i > ], calculated by offdiagonals
+    modes::M # Store the modes of addr
+end
+
+function starting_address(h::MolecularHamiltonian)
+    return h.starting_address
+end
+
+function operator_column(h::MolecularHamiltonian{T,A,D}, a::FermiFS2C)::MolecularHamiltonianOperatorColumn where {T<:Number,A,D}
+    diag = zero(T)
+    ods = Vector{Float64}[]
+
+    modes = modes_extract(a)
+
+    one_elec_int = one_electron_integral(h.fcidump.int1, modes.occupied)
+    two_elec_int = two_electron_integral(h.fcidump.int2, modes.occupied)
+    diag = h.fcidump.int0 + one_elec_int + two_elec_int
+
+    MolecularHamiltonianOperatorColumn{typeof(a),T,typeof(h),typeof(ods),typeof(modes)}(a, h, diag, ods, modes)
+end
+
+function diagonal_element(column::MolecularHamiltonianOperatorColumn{A,T,O,OD}) where {A<:FermiFS2C,T,O,OD}
+    return column.diag
+end
+
+function num_offdiagonals(column::MolecularHamiltonianOperatorColumn)
+    n_spin_orb = headvar(column.op.fcidump, "NORB") * 2
+    n_elec = headvar(column.op.fcidump, "NELEC")
+
+    # One-electron excitation
+    n_one_electron_excitation = n_elec * (n_spin_orb - n_elec)
+
+    # Two-electron excitation
+    n_two_electron_excitation = binomial(n_elec, 2) * binomial(n_spin_orb - n_elec, 2)
+
+    n_one_electron_excitation + n_two_electron_excitation
+end
+
+const ModeIndex = @NamedTuple{ch::Int, orb::Int}
+
+struct ModeTransition
+    old::ModeIndex
+    new::ModeIndex
+end
+
+mutable struct MolecularHamiltonianOffDiagonalsIterState
+    occ_ind::Int
+    unocc_ind::Int
+    spin_ch::Int
+end
+
+struct MolecularHamiltonianOffDiagonals
+    modes::Modes{FermiFSIndex}
+    transition::Vector{ModeTransition}
+end
+
+function Base.iterate(mhod::MolecularHamiltonianOffDiagonals, state::MolecularHamiltonianOffDiagonalsIterState=MolecularHamiltonianOffDiagonalsIterState(1, 1, 1))
+    # Only includes one-electron exctitation
+    if state.spin_ch == 3
+        return nothing
+    end
+    cur_trans = ModeTransition(
+        (state.spin_ch, mhod.modes.occupied[state.spin_ch][state.occ_ind].mode),
+        (state.spin_ch, mhod.modes.unoccupied[state.spin_ch][state.unocc_ind].mode)
+    )
+    println(cur_trans)
+    if state.unocc_ind < length(mhod.modes.unoccupied[state.spin_ch])
+        state.unocc_ind += 1
+    else
+        state.unocc_ind = 1
+        if state.occ_ind < length(mhod.modes.occupied[state.spin_ch])
+            state.occ_ind += 1
+        else
+            state.occ_ind = 1
+            if state.spin_ch <= 2
+                state.spin_ch += 1
+            end
+        end
+    end
+    return MolecularHamiltonianOffDiagonals(mhod.modes, [cur_trans]), state
```

Comment:
According to the interface `offdiagonals` should return an iterator over `Pair`s with a new address and the value of the matrix element. Thus, the `iterate` function needs to return a tuple of the form `(address => value, state)`, or `nothing` when it is exhausted.